src/parser/AstNode.ts

```diff
@@ -0,0 +1,28 @@
+import { Location } from "../lexer";
+
+/**
+ * ABC (that's minimally compliant with ESLint/ESTree) for all nodes within
```

Comment:
I assume "ABC" is ~"American Broadcasting Company"~  ~"easy as 1-2-3"~ "abstract base class"?